numpy/f2py/crackfortran.py

```diff
@@ -832,7 +832,7 @@ def appenddecl(decl, decl2, force=1):
         decl = {}
     if not decl2:
         return decl
-    if decl is decl2:
+    if decl == decl2:
```

Comment:
As I understand it `decl` and `decl2` have to be dictionaries (so `==` is fine) and the following `if` block will override `decl` values with that of `decl2` where relevant so if `==` is true it has no effect (but does no harm, just redundant)

The only case this is not true is for `setattrspec(decl, l, force)` is if `force` is true: if `decl2 == decl` then `decl['attrspec']` would be modified by having its values appended to it. I thought this was *not* desirable behaviour but you're right that someone more familiar with `f2py` might want to confirm this change.